src/rgw/rgw_auth_s3.cc

```diff
@@ -614,17 +616,14 @@ get_v4_canonical_headers(const req_info& info,
       }
     }
 
-    canonical_hdrs_map[token] = rgw_trim_whitespace(token_value);
+    canonical_hdrs_set.insert(
+	boost::algorithm::join(std::vector<std::string>(
```

Comment:
I would love to avoid `std::vector` and dynallocs here. Maybe `boost::containers::static_vector` could help. Anyway, this isn't even a minor.

Reply:
there's a `string_join_reserve()` in rgw_string.h that can do this without needing a container:

`string_join_reserve(':', token, rgw_trim_whitespace(token_value))`